homeassistant/components/prometheus/__init__.py

```diff
@@ -378,6 +378,31 @@ def _handle_input_number(self, state):
                 )
             metric.labels(**self._labels(state)).set(value)
 
+    def _handle_number(self, state):
```

Comment:
Instead of copy-pasting the handler, may I suggest moving shared code to a separate method and calling it from both handlers? Something like this:

```python
def _handle_input_number(self, state):
  self._numeric_handler(state, "input_number", "input number")

def _handle_number(self, state):
  self._numeric_handler(state, "number", "number")

def _numeric_handler(state, domain, title):
  if unit := self._unit_string(state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)):
    metric = self._metric(
      f"{domain}_state_{unit}",
      self.prometheus_cli.Gauge,
      f"State of the {title} measured in {unit}",
  )
  ...
```

(this applies to the code of the integration itself, it's fine to have duplicate similar code in unit tests)

Reply:
Thanks, what a blunder on my site. Fixed in 72ad1ec74f9286c92d2481c953a05ef49867b8e9 .